dom/events/Event-dispatch-detached-input-and-change.html

```diff
@@ -0,0 +1,226 @@
+  assert_false(changeEventFired);
+
+  t.done();
+}, `detached radio should not emit input or change events on dispatchEvent(new MouseEvent('click')).`);
+
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  document.body.appendChild(input);
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
+  t.done();
+}, 'attached checkbox should emit input and change events on click().');
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'radio';
+  document.body.appendChild(input);
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
+  t.done();
+}, 'attached radio should emit input and change events on click().');
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  document.body.appendChild(input);
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.dispatchEvent(new MouseEvent('click'));
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
+  t.done();
+}, `attached checkbox should emit input and change events on dispatchEvent(new MouseEvent('click')).`);
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'radio';
+  document.body.appendChild(input);
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.dispatchEvent(new MouseEvent('click'));
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
+  t.done();
+}, `attached radio should emit input and change events on dispatchEvent(new MouseEvent('click')).`);
+
+
+async_test(t => {
+  const input = document.createElement('input');
+  input.type = 'checkbox';
+  const shadowHost = document.createElement('div');
+  document.body.appendChild(shadowHost);
+  const shadowRoot = shadowHost.attachShadow({mode: 'open'});
+  shadowRoot.appendChild(input);
+
+  let inputEventFired = false;
+  input.addEventListener('input', () => inputEventFired = true);
+  let changeEventFired = false;
+  input.addEventListener('change', () => changeEventFired = true);
+  input.click();
+  assert_true(inputEventFired);
+  assert_true(changeEventFired);
+
+  input.remove();
```

Comment:
For these tests, it's not really necessary to remove both the input and the shadow host; removing the shadow host will remove its children too.